engine/transaction/record/record_outline.go

```diff
@@ -0,0 +1,132 @@
+package record
+
+import (
+	"context"
+
+	"github.com/bitcoin-sv/go-sdk/spv"
+	trx "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/bitcoin-sv/spv-wallet/conv"
+	"github.com/bitcoin-sv/spv-wallet/engine/database"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/engine/transaction"
+	txerrors "github.com/bitcoin-sv/spv-wallet/engine/transaction/errors"
+	"github.com/bitcoin-sv/spv-wallet/engine/transaction/outlines"
+	"github.com/bitcoin-sv/spv-wallet/models/bsv"
+	"github.com/bitcoin-sv/spv-wallet/models/transaction/bucket"
+)
+
+// RecordTransactionOutline will validate, broadcast and save a transaction outline
+func (s *Service) RecordTransactionOutline(ctx context.Context, outline *outlines.Transaction) error {
+	tx, err := trx.NewTransactionFromBEEFHex(outline.BEEF)
+	if err != nil {
+		return txerrors.ErrTxValidation.Wrap(err)
+	}
+
+	if ok, err := spv.VerifyScripts(tx); err != nil {
+		return txerrors.ErrTxValidation.Wrap(err)
+	} else if !ok {
+		return txerrors.ErrTxValidation
+	}
+
+	utxos, err := s.getTrackedUTXOsFromInputs(ctx, tx)
+	if err != nil {
+		return err
+	}
+
+	txID := tx.TxID().String()
+	for _, utxo := range utxos {
+		utxo.Spend(txID)
+	}
+
+	newOutputs, newDataRecords, err := s.processAnnotatedOutputs(tx, &outline.Annotations)
+	if err != nil {
+		return err
+	}
+
+	if err = s.broadcaster.Broadcast(ctx, tx); err != nil {
+		return txerrors.ErrTxBroadcast.Wrap(err)
+	}
+
+	txRecord := database.Transaction{
+		ID:       txID,
+		TxStatus: database.TxStatusBroadcasted,
+	}
+
+	upsertOutputs := append(newOutputs, utxos...) //newly created outputs and spent utxos
+	err = s.repo.SaveTX(ctx, &txRecord, upsertOutputs, newDataRecords)
+	if err != nil {
+		return txerrors.ErrSavingData.Wrap(err)
+	}
+
+	return nil
+}
+
+// getTrackedUTXOsFromInputs gets stored-in-our-database outputs used in provided tx
+// NOTE: The flow accepts transactions with "other/not-tracked" UTXOs,
+// if the untracked output is correctly unlocked by the input script we have no reason to block the transaction;
+// but only the tracked UTXOs will be marked as spent (and considered for future double-spending checks)
+func (s *Service) getTrackedUTXOsFromInputs(ctx context.Context, tx *trx.Transaction) ([]*database.Output, error) {
+	outpoints := func(yield func(outpoint bsv.Outpoint) bool) {
+		for _, input := range tx.Inputs {
+			yield(bsv.Outpoint{
+				TxID: input.SourceTXID.String(),
+				Vout: input.SourceTxOutIndex,
+			})
+		}
+	}
+	storedUTXOs, err := s.repo.GetOutputs(ctx, outpoints)
+	if err != nil {
+		return nil, txerrors.ErrGettingOutputs.Wrap(err)
+	}
+
+	for _, utxo := range storedUTXOs {
+		if utxo.IsSpent() {
+			return nil, txerrors.ErrUTXOSpent
```

Comment:
Maybe let's combine this 
`txerrors.ErrUTXOSpent` with some info which of proceeded UTXOs is already marked as spent?